pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -83,10 +83,11 @@ def __verify_train_loop_configuration(self, model: 'pl.LightningModule') -> None
 
         has_overriden_optimization_functions = trainer.overriden_optimizer_step or trainer.overriden_optimizer_zero_grad
         if has_overriden_optimization_functions and going_to_accumulate_grad_batches and automatic_optimization:
-            raise MisconfigurationException(
+            rank_zero_warn(
                 'When overriding `LightningModule` optimizer_step or optimizer_zero_grad,'
-                ' `accumulate_grad_batches` in `Trainer` should be 1.'
-                ' It ensures optimizer_step or optimizer_zero_grad are called on every batch.'
+                ' `accumulate_grad_batches` in `Trainer` is usually 1.'
+                ' If `accumulate_grad_batches` is not 1, optimizer_step and optimizer_zero_grad'
+                ' are not called on every batch (rather, they are called on every optimization step).'
```

Comment:
At this point, we know that `accumulate_grad_batches != 1` so we can reword the warning.

Reply:
I reworded it slightly already, but I'm not exactly sure what to say here. Perhaps: "accumulate_grad_batches is not 1, but optimizer_step or optimizer_zero_grad is overridden. Note that optimizer_step and optimizer_zero_grad will not be called on every batch."

Reply:
```python
When using `Trainer(accumulate_grad_batches != 1)` and overriding `LightningModule.optimizer_{step,zero_grad}`, the hooks will not be called on every batch (rather, they are called on every optimization step)
```